kura/org.eclipse.kura.util/src/main/java/org/eclipse/kura/util/message/store/SqlMessageStoreQueries.java

```diff
@@ -0,0 +1,276 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Eurotech and/or its affiliates and others
+ * 
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ * 
+ * Contributors:
+ *  Eurotech
+ *******************************************************************************/
+package org.eclipse.kura.util.message.store;
+
+import static java.util.Objects.requireNonNull;
+
+public final class SqlMessageStoreQueries {
+
+    private final String sqlCreateTable;
+    private final String sqlMessageCount;
+    private final String sqlStore;
+    private final String sqlGetMessage;
+    private final String sqlGetNextMessage;
+    private final String sqlSetPublishedQoS1;
+    private final String sqlSetPublishedQoS0;
+    private final String sqlSetConfirmed;
+    private final String sqlAllUnpublishedMessages;
+    private final String sqlAllInFlightMessages;
+    private final String sqlAllDroppedInFlightMessages;
+    private final String sqlUnpublishAllInFlightMessages;
+    private final String sqlDropAllInFlightMessages;
+    private final String sqlDeleteDroppedMessages;
+    private final String sqlDeleteConfirmedMessages;
+    private final String sqlDeletePublishedMessages;
+    private final String sqlCreateNextMessageIndex;
```

Comment:
These required "index" fields need to be set also if someone don't wants to invoke the "createIndexes" method. Is that on purpose?

Reply:
Currently all fields are checked to ensure that the user explicitly reviews them at least once and provide a value for each of them. If a field is not used, any value can be set to it.